docs/controls/controls.rst

```diff
@@ -0,0 +1,477 @@
+
+.. literalinclude:: src/interface-stack/main.qml
+    :lines: 1-4, 10-16, 36-41, 66-72, 84
+
+The home page, ``Home.qml`` consists of a ``Page``, which is an control element that support headers and footers. In this example we simply center a ``Label`` with the text *Home Screen* on the page.
+
+.. literalinclude:: src/interface-stack/Home.qml
+
+Returning back to the ``main.qml``, we now look at the drawer part. This is where the navigation to the pages begin. The active parts of the user interface is the ``ÌtemDelegate`` items. In the ``onClicked`` handler, the next page is pushed onto the ``stackView``.
+
+As shown in the code below, it possible to push either a ``Component`` or a reference to a specific QML file. Either way results in that a new instance is created and is pushed onto the stack.
+
+.. literalinclude:: src/interface-stack/main.qml
+    :lines: 4, 10-12, 38, 42-66, 73-
+
+The other half of the puzzle is the toolbar. The idea is that a back button is shown when the ``stackView`` contains more than one page, otherwise a menu button is shown. The logic for this can be seen on the ``text`` property where the ``"\u..."`` strings represents the unicode symbols that we need.
+
+In the ``onClicked`` handler, we can see that when there is more than one page on the stack, the stack is poped, i.e. the top page is removed. If the stack contains only one item, i.e. the home screen, the drawer is opened.
+
+Below the ``ToolBar``, you find a ``Label``. This element shows the title of each page in the center of the header.
+    
+.. literalinclude:: src/interface-stack/main.qml
+    :lines: 4, 10-12, 13, 17-36, 81-
+
+Now we've seen how to reach the *About* and *Profile* pages, but we also want to make it possible to reach the *Edit Profile* page from the *Profile* page. This is done via the ``Button`` on the *Profile* page. When the button is clicked, the ``EditProfile.qml`` file is pushed onto the ``StackView``.
+
+.. figure:: assets/interface-stack-profile.png
+
+    The profile page with the *Edit* button.
+
+.. literalinclude:: src/interface-stack/Profile.qml
+
+Side by Side Screens
+--------------------
+
+For this example we create a user interface consisting of three pages that the user can shift through. The pages are shown in the diagram below. This could be the interface of a health tracking app, tracking the current state, the user's statistics and the overall statistics.
+
+.. digraph:: side_by_side_screens
+
+    "Current" -> "Your Statistics"
+    "Your Statistics" -> "Community Statistics"
+    
+The illustration below shows how the *Current* page looks in the application. The title and text come from the page, while the ``PageIndicator`` (the three dots at the bottom) come from ``main.qml``. The page indicator shows the user which page that is currently active, helping when navigating.
```

Comment:
Maybe this is just me, but based on the format in the StackView section, I was expecting to see the type of control used mentioned by this point. It would make it nice and consistent, and give it a more.. natural flow. :)